lab_3_keywords_textrank/main.py

```diff
@@ -173,7 +194,22 @@ def extract_pairs(tokens: tuple[int, ...], window_length: int) -> Optional[tuple
     In case of corrupt input data, None is returned:
     tokens must not be empty, window lengths must be integer, window lengths cannot be less than 2.
     """
-    pass
+    if not (tokens and isinstance(window_length, int) and window_length >= 2):
+        return None
+    pairs = []
+    for idx, token in enumerate(tokens):
+        for i in range(1, window_length):
+            try:
+                pair = tokens[idx], tokens[idx + i]
+                if pair[0] != pair[1]:
+                    pairs.append(pair)
+            except IndexError:
```

Comment:
if you are aiming at grade higher than 7 I will ask to modify this part of the code

It is wonderful that you are able to catch and process exceptions, yet _passing_ them is considered to be a bad practice as it is virtually ignoring them 

so I would rather you come up with an algorithm that considers those cases